### PR TITLE
Homework5 completed

### PR DESCRIPTION
**Pull request by Innokentiy945**

### CODE REVIEW

File: src/main/java/hw5/Composite/AbstractPageComposite.java [hunk 1]
```diff
@@ -1,44 +1,44 @@
-package hw3.Composite;
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw5/Composite/AbstractPageComposite.java:1]: Why do you do not just copy files?

File: src/main/java/hw5/Composite/AbstractPageComposite.java [hunk 2]
```diff
@@ -1,44 +1,44 @@
...
+
+
+    @FindBy(css = ".sidebar-menu > li")
+    private List<WebElement> leftSectionMenuItems;
+
+    protected WebDriver driver;
+
+    protected WebDriverWait wait;
+
+    protected AbstractPageComposite(WebDriver driver) throws InterruptedException {
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw5/Composite/AbstractPageComposite.java:23]: Why do you through this exception?

File: src/main/java/hw5/Composite/Benefits.java [hunk 3]
```diff
@@ -10,17 +10,16 @@
 
 public class Benefits extends AbstractPageComposite {
 
+    public Benefits(WebDriver driver) throws InterruptedException {
+        super(driver);
+    }
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw5/Composite/Benefits.java:15]: Java Code Convention

File: src/main/java/hw5/Composite/Frame.java [hunk 4]
```diff
@@ -8,7 +8,7 @@
 
 public class Frame extends AbstractPageComposite {
 
-    public Frame(WebDriver driver) {
+    public Frame(WebDriver driver) throws InterruptedException {
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw5/Composite/Frame.java:11]: Java Code Convention
Why do you need this exception here?

File: src/main/java/hw5/Composite/HeaderMenu.java [hunk 5]
```diff
@@ -1,93 +1,93 @@
...
+    @FindBy(id = "user-name")
+    private WebElement userName;
+
+    @FindBy(css = ".m-l8 > li")
+    private List<WebElement> headerMenuElements;
+
+    @FindBy(css = ".m-l8 .dropdown")
+    private WebElement serviceMenu;
+
+
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw5/Composite/HeaderMenu.java:43]: One more time Java Code convention and exception

File: src/main/java/hw5/Pages/HomePage.java [hunk 6]
```diff
@@ -1,91 +1,91 @@
...
+import org.openqa.selenium.WebDriver;
+
+import java.util.List;
+
+public class HomePage extends AbstractPage {
+
+    protected Benefits benefits;
+    protected Frame frame;
+    protected LeftMenu leftMenu;
+    protected HeaderMenu headerMenu;
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw5/Pages/HomePage.java:13]: Do we have header and leftMenu only in the `HomePage`?

File: src/test/java/hw2/BaseTest.java [hunk 7]
```diff
@@ -1,59 +0,0 @@
-package hw2;
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw2/BaseTest.java:1]: Why do you delete hw2 from request?

File: src/test/java/hw3/BaseTest/TestBase.java [hunk 8]
```diff
@@ -1,52 +0,0 @@
-package hw3.BaseTest;
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw3/BaseTest/TestBase.java:1]: Why do you delete hw3 related files from the pull request?

File: src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+    protected WebDriver driver;
+    protected HomePage homePage;
+    protected DifferentElementPage differentElementPage;
+
+
+
+
+
+    public DifferentElementPageSteps(WebDriver driver) throws InterruptedException {
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java:25]: Why do you need this exception?

File: src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    public void goToDifferentElementPAge() {
+        homePage.goToDifferentPage();
+    }
+
+
+    @Step("Selecting checkboxes")
+    public void selectingCheckboxes() {
+        List<String> checkboxes = Arrays.asList("Water", "Wind", "Selen");
+        differentElementPage.defineButtonElements(checkboxes);
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java:42]: This step is incorrect! Which checkboxes you selected? Could you reproduce this step if you will get it from the report?

File: src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+
+    @Step("Selecting in dropdown Yellow")
+    public void selectingItemInDropdown() {
+        String color = "Yellow";
+        differentElementPage.defineColorDropDown(color);
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java:49]: Why value is hardcoded?

File: src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+
+    @Step("Verifiy that log is displayed after selecting")
+    public void verifyingIndividualLog() {
+        List<String> logs = Arrays.asList("Wind", "Water", "Selen", "Yellow");
+        differentElementPage.isElementInLogDisplayed(logs);
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java:56]: Which log do you check? How I should understand from the report which log is displayed?

File: src/test/java/hw5/StepsPattern/HomePageSteps.java [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+    public void openUrl() {
+        driver.get(url);
+    }
+
+    @Step("Verify site`s title")
+    public void checkTitle() {
+       String actualTitle = driver.getTitle();
+       String expectedTitle = "Home Page";
+       assertEquals(actualTitle, expectedTitle);
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/StepsPattern/HomePageSteps.java:41]: What is the title site

File: src/test/java/hw5/StepsPattern/HomePageSteps.java [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+    public HomePageSteps(WebDriver driver) throws InterruptedException {
+        this.driver = driver;
+        homePage = new HomePage(driver);
+    }
+
+
+    @Step("Test site opening")
+    public void openUrl() {
+        driver.get(url);
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/StepsPattern/HomePageSteps.java:34]: Which is site we open?

File: src/test/java/hw5/StepsPattern/HomePageSteps.java [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+       String expectedTitle = "Home Page";
+       assertEquals(actualTitle, expectedTitle);
+    }
+
+    @Step("User login on site")
+    public void logInOperation() {
+        homePage.logInOperation(PropertiesOpener.read("user"), PropertiesOpener.read("password"));
+        homePage.getNameOfUser();
+
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/StepsPattern/HomePageSteps.java:48]: Which is user logined?

File: src/test/java/hw5/StepsPattern/HomePageSteps.java [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+
+    @Step("Verifying user name")
+    public void assertUserName() {
+        homePage.isUserNameDisplayed();
+    }
+
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/StepsPattern/HomePageSteps.java:54]: ????? What is user name

File: src/test/java/hw5/StepsPattern/HomePageSteps.java [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+
+    @Step("Verifying header menu text")
+    public void verifyingHeaderMenuText() {
+        List<String> actualTextOfHeaderMenuItems = homePage.getHeaderMenuElements();
+        List<String> expectedTextOfHeaderMenuItems = Arrays.asList("HOME", "CONTACT FORM", "SERVICE", "METALS & COLORS");
+        assertEquals(actualTextOfHeaderMenuItems, expectedTextOfHeaderMenuItems);
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/StepsPattern/HomePageSteps.java:61]: Which header?

File: src/main/java/hw5/Composite/AbstractPageComposite.java (hunk 2 above)
Comment (Innokentiy945) [reply src/main/java/hw5/Composite/AbstractPageComposite.java]: Removed unusable exceptions in project.

File: src/main/java/hw5/Composite/Benefits.java (hunk 3 above)
Comment (Innokentiy945) [reply src/main/java/hw5/Composite/Benefits.java:15]: Constructors moved under variables in all classes.

File: src/main/java/hw5/Composite/HeaderMenu.java (hunk 5 above)
Comment (Innokentiy945) [reply src/main/java/hw5/Composite/HeaderMenu.java]: Constructors moved under variables in all classes and deleted all unusable exceptions.

File: src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java (hunk 9 above)
Comment (Innokentiy945) [reply src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java]: Removed unusable exceptions in all classes

File: src/test/java/hw5/StepsPattern/HomePageSteps.java (hunk 15 above)
Comment (Innokentiy945) [reply src/test/java/hw5/StepsPattern/HomePageSteps.java:48]: Rewrite step to: "Home page opening"

File: src/test/java/hw5/StepsPattern/HomePageSteps.java (hunk 14 above)
Comment (Innokentiy945) [reply src/test/java/hw5/StepsPattern/HomePageSteps.java:34]: Rewrite step to: "Home page opening"

File: src/test/java/hw5/StepsPattern/HomePageSteps.java (hunk 16 above)
Comment (Innokentiy945) [reply src/test/java/hw5/StepsPattern/HomePageSteps.java:54]: Rewrite step to: "Verifying logged in user name"

File: src/test/java/hw5/StepsPattern/HomePageSteps.java (hunk 17 above)
Comment (Innokentiy945) [reply src/test/java/hw5/StepsPattern/HomePageSteps.java:61]: Rewrite step to: "Verifying home page`s header menu text"

File: src/test/java/hw5/StepsPattern/HomePageSteps.java (hunk 13 above)
Comment (Innokentiy945) [reply src/test/java/hw5/StepsPattern/HomePageSteps.java:41]: Rewrited and removed all hardcode from steps classes.

File: src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java (hunk 11 above)
Comment (Innokentiy945) [reply src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java]: Replaced with parameterized methods in all cases.

File: src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java (hunk 12 above)
Comment (Innokentiy945) [reply src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java]: Fixed with multiple log for each elements.

File: src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java (hunk 10 above)
Comment (Innokentiy945) [reply src/test/java/hw5/StepsPattern/DifferentElementPageSteps.java]: Rewrited one method for multiple.

File: src/main/java/hw5/Composite/Frame.java (hunk 4 above)
Comment (Innokentiy945) [reply src/main/java/hw5/Composite/Frame.java:11]: Removed unusable exceptions in project and constructors moved under variables in all classes

File: src/main/java/hw5/Pages/HomePage.java (hunk 6 above)
Comment (Innokentiy945) [reply src/main/java/hw5/Pages/HomePage.java]: Replaced with public modificator

File: src/test/java/hw3/BaseTest/TestBase.java (hunk 8 above)
Comment (Innokentiy945) [reply src/test/java/hw3/BaseTest/TestBase.java:1]: I create new origin branch from master, then did pull to my local branch. After this i deleted unuseful directories.

File: src/test/java/hw2/BaseTest.java (hunk 7 above)
Comment (Innokentiy945) [reply src/test/java/hw2/BaseTest.java:1]: I create new origin branch from master, then did pull to my local branch. After this i deleted unuseful directories.

File: src/test/java/hw3/BaseTest/TestBase.java (hunk 8 above)
Comment (DmitryKhodakovsky) [reply src/test/java/hw3/BaseTest/TestBase.java:1]: Why do decide that these are unuseful directories?

**Pull request merged**